transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/logging/WsebAcceptorUserLoggingMultiplePrincipalsInConfigIT.java

```diff
@@ -0,0 +1,140 @@
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.kaazing.test.util.MemoryAppender;
+import org.kaazing.test.util.MethodExecutionTrace;
+
+/**
+ * WsebAcceptorUserLoggingIT - verifies that if multiple Principals are defined in the config
+ * only the first one is used
+ *
+ * Logging on different layers:
+ * - TCP: hostname:port
+ * - HTTP: principal hostname:port
+ * - WSEB: principal hostname:port
+ */
+public class WsebAcceptorUserLoggingMultiplePrincipalsInConfigIT {
+    private static final String ROLE = "USER";
+    private static final String DEMO_REALM = "demo";
+    private static final String TEST_PRINCIPAL_PASS = "testPrincipalPass";
+    private static final String TEST_PRINCIPAL_NAME = "testPrincipalName";
+    private List<String> expectedPatterns;
+    private List<String> forbiddenPatterns;
+    private final K3poRule k3po = new K3poRule();
+    private TestRule checkLogMessageRule = new TestRule() {
+        @Override
+        public Statement apply(final Statement base, Description description) {
+            return new Statement() {
+                @Override
+                public void evaluate() throws Throwable {
+                    base.evaluate();
+                    MemoryAppender.assertMessagesLogged(expectedPatterns,
+                            forbiddenPatterns, ".*\\[.*#.*].*", true);
+                }
+            };
+        }
+    };
+    public GatewayRule gateway = new GatewayRule() {
+        {
+        GatewayConfiguration configuration = new GatewayConfigurationBuilder()
+                .property(WSE_SPECIFICATION.getPropertyName(), "true")
+                .service()
+                    .accept("ws://localhost:8080/path")
+                    .type("echo")
+                    .crossOrigin()
+                        .allowOrigin("http://localhost:8001")
+                    .done()
+                    .realmName(DEMO_REALM)
+                        .authorization()
+                        .requireRole(ROLE)
+                    .done()
+                .done()
+                .security()
+                    .realm()
+                        .name(DEMO_REALM)
+                        .description("Kaazing WebSocket Gateway Demo")
+                        .httpChallengeScheme("Basic")
+                        .userPrincipalClass("org.kaazing.gateway.security.auth.config.parse.DefaultUserConfig")
+                        .userPrincipalClass("org.kaazing.gateway.security.auth.config.parse.TestConfig")
```

Comment:
We should probably have TestConfig (or some other Principal implementation as per the above [comment](https://github.com/kaazing/gateway/pull/450/files#r56053184)) on the subject within the login module commit method. We can create a new login module extending BasicLoginModuleWithDefaultUserConfig, similarly to BasicLoginModuleWithMultiplePrincipalsOnTheSubject.